### ISSUE TITLE
Citations button not working

### ISSUE THREAD

**Issue by laurenwalker** [NCEAS/metacatui#1254]
When I click on the citations button on a dataset, I get this error:

```
Uncaught TypeError: Cannot read property 'getTime' of null
    at MetricsChartView.js?v=2.8.4:146
    at Array.forEach (<anonymous>)
    at n.render (MetricsChartView.js?v=2.8.4:145)
    at n.drawMetricsChart (MetricModalView.js?v=2.8.4:192)
    at n.render (MetricModalView.js?v=2.8.4:64)
    at n.showMetricModal (MetricView.js?v=2.8.4:80)
    at HTMLAnchorElement.dispatch (jquery-1.9.1.min.js?v=2.8.4:3)
    at HTMLAnchorElement.v.handle (jquery-1.9.1.min.js?v=2.8.4:3)
```

Example dataset: https://arcticdata.io/catalog/view/doi:10.18739/A2804XJ9V

It looks like we don't have any `try, catch` statements in the MetricsChartView. At the very least, the `render()` function should be wrapped in one, and as much as reasonable, chunks of code should be wrapped in `try`s as well so as much of the view will display as possible. But fixing the bug should be the first priority.

**Comment by rushirajnenuji** [comment]
Thanks for the directions, Lauren. This was one of the manually added citations, so the `linkPublicationDate` was NULL (date when a citation relation was established between the two resources). Metrics Service already handles this edge case, but we haven't made it to production yet. Hence we encountered this bug in the UI. 

Fixed this issue from the UI side (refer above commit). For cases when we do not have a link Publication date we fall back to the current year and month.
